packages/flutter/test/widgets/overflow_bar_test.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+void main() {
+  testWidgets('OverflowBar documented defaults', (WidgetTester tester) async {
+    final OverflowBar bar = OverflowBar();
+    expect(bar.spacing, 0);
+    expect(bar.overflowSpacing, 0);
+    expect(bar.overflowDirection, VerticalDirection.down);
+    expect(bar.textDirection, null);
+    expect(bar.clipBehavior, Clip.none);
+    expect(bar.children, const <Widget>[]);
+  });
+
+  testWidgets('EmptyOverflowBar', (WidgetTester tester) async {
+    const Size size = Size(16, 24);
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Center(
+          child: ConstrainedBox(
+            constraints: BoxConstraints.tight(size),
+            child: OverflowBar(),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(OverflowBar)), size);
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Center(
+          child: OverflowBar(),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byType(OverflowBar)), Size.zero);
+  });
+
+  testWidgets('OverflowBar horizontal layout', (WidgetTester tester) async {
+    final Key child1Key = UniqueKey();
+    final Key child2Key = UniqueKey();
+    final Key child3Key = UniqueKey();
+
+    Widget buildFrame({ double spacing, TextDirection textDirection }) {
+      return Directionality(
+        textDirection: textDirection,
+        child: Align(
+          alignment: Alignment.topLeft,
+          child: OverflowBar(
+            spacing: spacing,
+            children: <Widget>[
+              SizedBox(width: 48, height: 48, key: child1Key),
+              SizedBox(width: 64, height: 64, key: child2Key),
+              SizedBox(width: 32, height: 32, key: child3Key),
+            ],
+          ),
+        ),
+      );
+    }
+
+    // Children are vertically centered, start at x=0
+    await tester.pumpWidget(buildFrame(spacing: 0, textDirection: TextDirection.ltr));
+    expect(tester.getRect(find.byKey(child1Key)), const Rect.fromLTRB(0, 8, 48, 56));
+    expect(tester.getRect(find.byKey(child2Key)), const Rect.fromLTRB(48, 0, 112, 64));
+    expect(tester.getRect(find.byKey(child3Key)), const Rect.fromLTRB(112, 16, 144, 48));
+
+    // Children are vertically centered, start at x=0
+    await tester.pumpWidget(buildFrame(spacing: 10, textDirection: TextDirection.ltr));
+    expect(tester.getRect(find.byKey(child1Key)), const Rect.fromLTRB(0, 8, 48, 56));
+    expect(tester.getRect(find.byKey(child2Key)), const Rect.fromLTRB(10.0 + 48, 0, 10.0 + 112, 64));
```

Comment:
Nit: Should probably be consistent with keeping/omitting the `.0` for doubles.

Reply:
I find that omitting the `.0` makes code easier to read however it was necessary to introduce one in these expressions so that they eval to doubles instead of ints. I realize that it's inconsistent but I've tried to be consistent about that.